File: ClientBridge/src/main/php/client/packets/collection/other/PacketPing.php
<?php
/**
 * Created by Nick on 24 okt 2019.
 * Copyright © ImSpooks
 */

namespace client\packets\collection\other;

use client\packets\init\Packet;
use client\packets\init\channels\WrappedInputStream;
use client\packets\init\channels\WrappedOutputStream;

class PacketPing extends Packet {

    /** @var int */ 	private $clientTime;

    public function __construct(int $clientTime = 0) {
        $this->clientTime = $clientTime;
    }

    public function receive(WrappedInputStream $in) {
        $this->clientTime = $in->readLong();
    }

    public function send(WrappedOutputStream $out) {
        $out->writeLong($this->clientTime);
    }

    public function getClientTime(): int {
        return $this->clientTime;
    }

}

File: Packets/src/main/java/me/ImSpooks/core/packets/collection/other/PacketPing.java
package me.ImSpooks.core.packets.collection.other;

import lombok.Getter;
import me.ImSpooks.core.packets.collection.GlobalPacket;
import me.ImSpooks.core.packets.init.Packet;
import me.ImSpooks.core.packets.init.channels.WrappedInputStream;
import me.ImSpooks.core.packets.init.channels.WrappedOutputStream;

import java.io.IOException;

/**
 * Created by Nick on 17 okt. 2019.
 * Copyright © ImSpooks
 */
@GlobalPacket
public class PacketPing extends Packet {

    @Getter private long clientTime;

    public PacketPing(long clientTime) {
        this.clientTime = clientTime;
    }

    public PacketPing() {
    }

    @Override
    public void send(WrappedOutputStream out) throws IOException {
        out.writeLong(this.clientTime);
    }

    @Override
    public void receive(WrappedInputStream in) throws IOException {
        this.clientTime = in.readLong();
    }
}
